arkouda/strings.py

```diff
@@ -255,6 +260,146 @@ def get_lengths(self) -> pdarray:
                         format(self.objtype, self.offsets.name, self.bytes.name)
         return create_pdarray(generic_msg(cmd=cmd,args=args))
 
+    def _refresh_regex_dict(self):
+        """
+        Internal function to recreate regex_dict without any stale pdarrays (that are no longer in the symbol table)
+        """
+        sym_tab = list_symbol_table()
+        self.regex_dict = {key: val for key, val in self.regex_dict.items() if all([pda.name in sym_tab for pda in val])}
+
+    def print_cached_regex_patterns(self):
+        """
+        Prints the regex patterns for which Strings.find(pattern) has been cached
+        """
+        self._refresh_regex_dict()
+        print(self.regex_dict.keys())
```

Comment:
It looks like `_refresh_regex_dict` is only called by `print_cached_regex_patterns`, I'm wondering if a better approach would be to return a copy of the dictionary you made in `_refresh_regex_dict` if you're worried about a user modifying the state of it.  Then the user can choose to print it or not... having a function that prints feels weird to me.

Reply:
ooo yeah i agree! that makes more sense